mlir/lib/Dialect/Utils/ReshapeOpsUtils.cpp

```diff
@@ -8,6 +8,7 @@
 
 #include "mlir/Dialect/Utils/ReshapeOpsUtils.h"
 
+#include "mlir/Dialect/Arith/IR/Arith.h"
 #include "mlir/IR/AffineMap.h"
 #include "mlir/IR/Builders.h"
```

Comment:
Is this dependency from `Dialect/Utils/` onto the Arith dialect acceptable, or should this new `inferExpandShapeOutputShape()` function be moved to somewhere else (`Dialect/Arith/Utils`? Or memref _and_ tensor where the code is used, with some code duplication?)

Reply:
I tried moving it to tensor/utils, but there seem to be circular dependencies. I can handle this in the next PR. Can we go ahead for this patch as it is?

Reply:
+1, see my other comment, thanks for deferring further improvements to follow-up PRs!